smithy-linters/src/main/java/software/amazon/smithy/linters/ShouldHaveUsedTimestampValidator.java

```diff
@@ -111,15 +108,27 @@ private ShouldHaveUsedTimestampValidator(Config config) {
 
     @Override
     public List<ValidationEvent> validate(Model model) {
-        ShapeVisitor<List<ValidationEvent>> visitor = Shape.<List<ValidationEvent>>visitor()
-                .when(StringShape.class, s -> validateSimpleShape(s, patterns))
-                .when(ShortShape.class, s -> validateSimpleShape(s, patterns))
-                .when(IntegerShape.class, s -> validateSimpleShape(s, patterns))
-                .when(LongShape.class, s -> validateSimpleShape(s, patterns))
-                .when(FloatShape.class, s -> validateSimpleShape(s, patterns))
-                .when(StructureShape.class, shape -> validateStructure(shape, model, patterns))
-                .when(UnionShape.class, shape -> validateUnion(shape, model, patterns))
-                .orElse(ListUtils.of());
+        ShapeVisitor<List<ValidationEvent>> visitor = new ShapeVisitor.Default<List<ValidationEvent>>() {
+            @Override
+            protected List<ValidationEvent> getDefault(Shape shape) {
+                if (shape.isStringShape() || shape instanceof NumberShape) {
```

Comment:
The expansion on this to be `NumberShape` means a few more shape types will be validated: `byte`, `double`, and the `Big*` numbers. Seems like that would be intended but calling out in case.